### PR TITLE
Adding pyextremes

### PR DESCRIPTION
**Pull request by georgebv**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by georgebv** [comment]
@conda-forge-admin, please ping team @conda-forge/help-python

Please review my recipe, all tests and checks have passed.

**Comment by georgebv** [comment]
Please review my recipe. The force-push tag above is there because I revised my recipe based on comments from other PR's and squashed everything into a single comment.

@conda-forge-admin, please ping team

File: recipes/pyextremes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - numpy >=1.18.1
+    - scipy >=1.4.1
+    - pandas >=1.0.1
+    - matplotlib-base >=3.1.0
+    - emcee >=2.2.1
```
Comment (scopatz) [inline recipes/pyextremes/meta.yaml:25]: Most of these are probably not host requirements needed to build the package. Should be just python and pip

**Comment by georgebv** [comment]
@scopatz tried removing these and immediately got an error. If I am not including these in the host section then those packages are not installed and my package cannot get imported. See this:

```
The following NEW packages will be INSTALLED:

    ca-certificates: 2020.4.5.2-hecda079_0     conda-forge
    certifi:         2020.4.5.1-py38h32f6830_0 conda-forge
    libcxx:          10.0.0-h1af66ff_2         conda-forge
    libffi:          3.2.1-h4a8c4bd_1007       conda-forge
    ncurses:         6.1-h0a44026_1002         conda-forge
    openssl:         1.1.1g-h0b31af3_0         conda-forge
    pip:             20.1.1-py_1               conda-forge
    python:          3.8.3-cpython_hd5f0129_0  conda-forge
    python_abi:      3.8-1_cp38                conda-forge
    readline:        8.0-hcfe32e1_0            conda-forge
    setuptools:      47.1.1-py38h32f6830_0     conda-forge
    sqlite:          3.30.1-h93121df_0         conda-forge
    tk:              8.6.10-hbbe82c9_0         conda-forge
    wheel:           0.34.2-py_1               conda-forge
    xz:              5.2.5-h0b31af3_0          conda-forge
    zlib:            1.2.11-h0b31af3_1006      conda-forge
```

No matplotlib or numpy above. Then I'm getting this error:

```
  Created temporary directory: /private/tmp/pip-req-build-c0tfuzkx
  Added file://$SRC_DIR to build tracker '/private/tmp/pip-req-tracker-crmq2frm'
    Running setup.py (path:/private/tmp/pip-req-build-c0tfuzkx/setup.py) egg_info for package from file://$SRC_DIR
    Created temporary directory: /private/tmp/pip-pip-egg-info-podp_sl5
    Running command python setup.py egg_info
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-req-build-c0tfuzkx/setup.py", line 2, in <module>
        import pyextremes
      File "/private/tmp/pip-req-build-c0tfuzkx/pyextremes/__init__.py", line 17, in <module>
        from pyextremes.eva import EVA
      File "/private/tmp/pip-req-build-c0tfuzkx/pyextremes/eva.py", line 22, in <module>
        import matplotlib.gridspec
    ModuleNotFoundError: No module named 'matplotlib'
```

Had to add them back for the build to work.

**Comment by georgebv** [comment]
@scopatz I found out why this is happening - my package `pyextremes` was imported within `setup.py` to get version from `__init__`. I guess this could cause problems with conda if it would try installing my package before its other dependencies. Updated this behavior, should be OK now.

**Comment by georgebv** [comment]
@conda-forge-admin, please ping team

**Pull request closed without merging**